### ISSUE TITLE
dodgy memsets

### ISSUE THREAD

**Issue by loveshack** [intel/mpi-benchmarks#32]
GCC issues these warnings with the 2019.6 release which look valid:

```
helpers/helper_IMB_functions.h:594:40: warning: 'memset' used with length equal to number of elements without multiplication by element size [-Wmemset-elt-size]
             memset(time, 0, MAX_TIME_ID);
                                        ^
```

```
../src_c/IMB_mem_manager.c:707:9: warning: 'memset' used with length equal to number of elements without multiplication by element size [-Wmemset-elt-size]
         memset(time, 0, MAX_TIME_ID);
         ^~~~~~
```

**Comment by VinnitskiV** [comment]
@loveshack Thank you for your contribution! 
This issue will be fixed in one of next releases.

**Comment by a-v-medvedev** [comment]
ok it's December 2021, the issue is still there...

**Comment by jeffhammond** [comment]
It's 2023 and still there.  Here's the fix.
```patch
diff --git a/src_cpp/helpers/helper_IMB_functions.h b/src_cpp/helpers/helper_IMB_functions.h
index d462b48..8c0672b 100644
--- a/src_cpp/helpers/helper_IMB_functions.h
+++ b/src_cpp/helpers/helper_IMB_functions.h
@@ -591,7 +591,7 @@ struct Bmark_descr {
             int acc_rep_test, t_sample;
             int selected_n_sample = ITERATIONS->n_sample;

-            memset(time, 0, MAX_TIME_ID);
+            memset(time, 0, MAX_TIME_ID*sizeof(double));
             if (iter == 0 || BMODE->type == Sync) {
                 ITERATIONS->n_sample_prev = ITERATIONS->msgspersample;
                 if (c_info->n_lens > 0) {
```

**Comment by JuliaRS** [comment]
@jeffhammond, @loveshack  thank you for your contribution,
I am going to add it to next release of IMB.

**Comment by JuliaRS** [comment]
For src_cpp/helpers/helper_IMB_functions.h it was fixed:
https://github.com/intel/mpi-benchmarks/blob/master/src_cpp/helpers/helper_IMB_functions.h#L592

For src_c/IMB_mem_manager.c it will be fix in IMB 2021.8 version.
I am going to close this issues
